<!-- -*- mode: XML -*- -->

<launch>
  <arg name="master_ip" default="localhost"/>
  <arg name="remote" default="false"/>

  <machine name="master" address="$(arg master_ip)" env-loader="/home/jdelacruz/rosenv.sh" user="jdelacruz" />

    <node if="$(arg remote)" machine="master" name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                pan_tilt_port:
                    port_name: "/dev/ttyUSB1"
                    baud_rate: 1000000
                    min_motor_id: 1
                    max_motor_id: 25
                    update_rate: 40
        </rosparam>
    </node>

    <node unless="$(arg remote)" name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                pan_tilt_port:
                    port_name: "/dev/ttyUSB0"
                    baud_rate: 1000000
                    min_motor_id: 1
                    max_motor_id: 25
                    update_rate: 40
        </rosparam>
    </node>
</launch>
